luaui/Widgets/gui_sensor_ranges_los.lua

```diff
@@ -246,44 +189,45 @@ function widget:Shutdown()
 	WG.losrange = nil
 end
 
-function widget:UnitDestroyed(unitID, unitDefID, unitTeam)
-	if unitList[unitID] then
-		unitList[unitID] = nil
-		popElementInstance(circleInstanceVBO, unitID)
-	end
-end
+function widget:VisibleUnitAdded(unitID, unitDefID, unitTeam, noupload)
+	--Spring.Echo("widget:VisibleUnitAdded",unitID, unitDefID, unitTeam, noupload)
+	unitTeam = unitTeam or spGetUnitTeam(unitID)
+	noupload = noupload == true
+	if unitRange[unitDefID] == nil or unitTeam == gaiaTeamID then return end
+	
+	if (not (spec and fullview)) and (not spIsUnitAllied(unitID)) then -- given units are still considered allies :/
+		return
+	end -- display mode for specs
+	
+	local _, _, _, _, buildProgress = Spring.GetUnitHealth(unitID)
+	if buildProgress < 0.99 then return end
```

Comment:
The API is event driven, an one important thing to note, is when a unit is finished building (UnitFinished), then VisibleUnitRemoved is called, and then visibleUnitAdded again. Because a lot of things change when a unit finishes construction.